agents/PCS-Large-LP-Deposit-Withdrawal/src/pool.fetcher.ts

```diff
@@ -0,0 +1,56 @@
+import { Contract, BigNumber, providers } from "ethers";
+import { FUNCTIONS_ABI } from "./constants";
+import LRU from "lru-cache";
+
+export default class PoolFetcher {
+  private provider: providers.Provider;
+  private cache: LRU<string, any[]>;
```

Comment:
Why do you need a cache in here? How is it helping? I don't really see a necessity for this. If the data for the same block were being queried multiple times, would have made sense, but don't really see that happening.